bnd-maven-plugin ConcurrentModificationException on latest OpenJDK 15 EA

I've been using the `bnd-maven-plugin` for a long time with no issues and have not made any recent configuration changes.  Beginning tonight, the Travis CI build is failing on the OpenJDK15 (early access) build with this error:

> [ERROR] Failed to execute goal biz.aQute.bnd:bnd-maven-plugin:5.0.1:bnd-process (default) on project oshi-core: bnd error: null: ConcurrentModificationException

The Java version used is:

```
openjdk version "15-ea" 2020-09-15
OpenJDK Runtime Environment (build 15-ea+18-776)
OpenJDK 64-Bit Server VM (build 15-ea+18-776, mixed mode, sharing)
```
Link to failing Travis CI job: https://travis-ci.org/github/oshi/oshi/jobs/673288964

It worked as recently as a two days ago on a slightly earlier EA build:
```
openjdk version "15-ea" 2020-09-15
OpenJDK Runtime Environment (build 15-ea+17-717)
OpenJDK 64-Bit Server VM (build 15-ea+17-717, mixed mode, sharing)
```
Link to passing Travis CI job from 2 days ago: https://travis-ci.org/github/oshi/oshi/jobs/672568551

Other relevant version strings:
```
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /home/travis/.m2/wrapper/dists/apache-maven-3.6.3-bin/1iopthnavndlasol9gbrbg6bf2/apache-maven-3.6.3
Java version: 15-ea, vendor: Oracle Corporation, runtime: /home/travis/openjdk15
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "4.15.0-1055-gcp", arch: "amd64", family: "unix"
```

Relevant portions of my POM configuration:
```
<properties>
  <bnd-maven-plugin.version>5.0.1</bnd-maven-plugin.version>
</properties>

<build>
  <pluginManagement>
    <plugins>
      <plugin>
        <groupId>biz.aQute.bnd</groupId>
        <artifactId>bnd-maven-plugin</artifactId>
        <version>${bnd-maven-plugin.version}</version>
        <configuration>
          <bnd><![CDATA[
          Export-Package: oshi.*;-noimport:=true;-split-package:=merge-first
          Bundle-SymbolicName: ${project.groupId}.${project.artifactId}
          -snapshot: SNAPSHOT
        ]]></bnd>
        </configuration>
      </plugin>
    </plugins>
  </pluginManagement>
  <plugins>
    <plugin>
      <groupId>biz.aQute.bnd</groupId>
      <artifactId>bnd-maven-plugin</artifactId>
      <executions>
        <execution>
          <goals>
            <goal>bnd-process</goal>
          </goals>
        </execution>
      </executions>
    </plugin>
  </plugins>
</build>
```

Happy to help troubleshoot but I have no idea where to even start!

Figuring how to add the -X to the Travis install script was an adventure, but I finally got it. :)

Link to the build and full log is here: https://travis-ci.org/github/oshi/oshi/jobs/673301183

Relevant (?) trace:
```
[INFO] --- bnd-maven-plugin:5.0.1:bnd-process (default) @ oshi-core ---
[DEBUG] Configuring mojo biz.aQute.bnd:bnd-maven-plugin:5.0.1:bnd-process from plugin realm ClassRealm[plugin>biz.aQute.bnd:bnd-maven-plugin:5.0.1, parent: java.net.URLClassLoader@37918c79]
[DEBUG] Configuring mojo 'biz.aQute.bnd:bnd-maven-plugin:5.0.1:bnd-process' with basic configurator -->
[DEBUG]   (f) bnd = Export-Package: oshi.*;-noimport:=true;-split-package:=merge-first
						Bundle-SymbolicName: com.github.oshi.oshi-core
						-snapshot: SNAPSHOT
[DEBUG]   (f) bndfile = bnd.bnd
[DEBUG]   (f) classesDir = /home/travis/build/oshi/oshi/oshi-core/target/classes
[DEBUG]   (f) includeClassesDir = true
[DEBUG]   (f) manifestPath = /home/travis/build/oshi/oshi/oshi-core/target/classes/META-INF/MANIFEST.MF
[DEBUG]   (f) mojoExecution = biz.aQute.bnd:bnd-maven-plugin:5.0.1:bnd-process {execution: default}
[DEBUG]   (f) outputDir = /home/travis/build/oshi/oshi/oshi-core/target/classes
[DEBUG]   (f) packagingTypes = [jar, war]
[DEBUG]   (f) project = MavenProject: com.github.oshi:oshi-core:5.0.0-SNAPSHOT @ /home/travis/build/oshi/oshi/oshi-core/pom.xml
[DEBUG]   (f) resources = [Resource {targetPath: null, filtering: false, FileSet {directory: /home/travis/build/oshi/oshi/oshi-core/src/main/resources, PatternSet [includes: {}, excludes: {}]}}]
[DEBUG]   (f) settings = org.apache.maven.execution.SettingsAdapter@6993c8df
[DEBUG]   (f) skip = false
[DEBUG]   (f) sourceDir = /home/travis/build/oshi/oshi/oshi-core/src/main/java
[DEBUG]   (f) targetDir = /home/travis/build/oshi/oshi/oshi-core/target
[DEBUG]   (f) warOutputDir = /home/travis/build/oshi/oshi/oshi-core/target/oshi-core-5.0.0-SNAPSHOT
[DEBUG] -- end configuration --
[DEBUG] loading bnd properties from bnd element in pom: MavenProject: com.github.oshi:oshi-parent:5.0.0-SNAPSHOT @ /home/travis/build/oshi/oshi/pom.xml
[DEBUG] loading bnd properties from bnd element in pom: MavenProject: com.github.oshi:oshi-core:5.0.0-SNAPSHOT @ /home/travis/build/oshi/oshi/oshi-core/pom.xml

<snip>

[ERROR] Failed to execute goal biz.aQute.bnd:bnd-maven-plugin:5.0.1:bnd-process (default) on project oshi-core: bnd error: null: ConcurrentModificationException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal biz.aQute.bnd:bnd-maven-plugin:5.0.1:bnd-process (default) on project oshi-core: bnd error: null
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:564)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:564)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:39)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:122)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:61)
Caused by: org.apache.maven.plugin.MojoExecutionException: bnd error: null
    at aQute.bnd.maven.plugin.AbstractBndMavenPlugin.execute (AbstractBndMavenPlugin.java:451)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:564)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:564)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:39)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:122)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:61)
Caused by: java.util.ConcurrentModificationException
    at java.util.TreeMap.callMappingFunctionWithCheck (TreeMap.java:742)
    at java.util.TreeMap.computeIfAbsent (TreeMap.java:596)
    at aQute.bnd.osgi.Jar.putResource (Jar.java:337)
    at aQute.bnd.osgi.Jar$2.visitFile (Jar.java:252)
    at aQute.bnd.osgi.Jar$2.visitFile (Jar.java:227)
    at java.nio.file.Files.walkFileTree (Files.java:2804)
    at aQute.bnd.osgi.Jar.buildFromDirectory (Jar.java:226)
    at aQute.bnd.osgi.Jar.<init> (Jar.java:128)
    at aQute.bnd.osgi.Jar.<init> (Jar.java:198)
    at aQute.bnd.osgi.Analyzer.addClasspath (Analyzer.java:2471)
    at aQute.bnd.maven.plugin.AbstractBndMavenPlugin.execute (AbstractBndMavenPlugin.java:190)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:564)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:564)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:39)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:122)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:61)
```

--- a/biz.aQute.bndlib.tests/test/test/JarTest.java
+++ b/biz.aQute.bndlib.tests/test/test/JarTest.java
@@ -3,17 +3,27 @@ package test;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.catchThrowable;
 import static org.assertj.core.data.Offset.strictOffset;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
 import java.io.ByteArrayInputStream;
 import java.io.ByteArrayOutputStream;
 import java.io.File;
 import java.io.IOException;
 import java.io.UncheckedIOException;
+import java.lang.reflect.Method;
 import java.util.jar.JarInputStream;
 import java.util.jar.Manifest;
 import java.util.zip.ZipEntry;
 import java.util.zip.ZipException;
 import java.util.zip.ZipInputStream;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
 
 import aQute.bnd.osgi.Builder;
 import aQute.bnd.osgi.Constants;
@@ -22,25 +32,22 @@ import aQute.bnd.osgi.FileResource;
 import aQute.bnd.osgi.Jar;
 import aQute.bnd.osgi.Resource;
 import aQute.lib.io.IO;
-import junit.framework.TestCase;
 
-@SuppressWarnings("resource")
 
-public class JarTest extends TestCase {
+public class JarTest {
 
 	File tmp;
 
-	private String getTestName() {
-		return getClass().getName() + "/" + getName();
-	}
-
-	@Override
-	protected void setUp() throws Exception {
-		tmp = IO.getFile("generated/tmp/test/" + getTestName());
+	@BeforeEach
+	public void setUp(TestInfo info) throws Exception {
+		Method testMethod = info.getTestMethod()
+			.get();
+		tmp = new File("generated/tmp/test/" + getClass().getName() + "/" + testMethod.getName()).getAbsoluteFile();
 		IO.delete(tmp);
 		IO.mkdirs(tmp);
 	}
 
+	@Test
 	public void testDeletePrefix() {
 		Resource r = new EmbeddedResource(new byte[1], 0L);
 
@@ -68,6 +75,7 @@ public class JarTest extends TestCase {
 		}
 	}
 
+	@Test
 	public void testDeleteSubDirs() {
 		Resource r = new EmbeddedResource(new byte[1], 0L);
 
@@ -97,6 +105,7 @@ public class JarTest extends TestCase {
 		}
 	}
 
+	@Test
 	public void testDeleteSubDirs2() {
 		Resource r = new EmbeddedResource(new byte[1], 0L);
 
@@ -124,6 +133,36 @@ public class JarTest extends TestCase {
 		}
 	}
 
+	@Test
+	public void testPackages() {
+		Resource r = new EmbeddedResource(new byte[1], 0L);
+
+		try (Jar jar = new Jar("test")) {
+			jar.putResource("com/example/foo/bar/x.class", r);
+			jar.putResource("com/example/foo/bar/package-info.class", r);
+
+			assertThat(jar.getDirectories()).containsKeys("com/example/foo/bar", "com/example/foo", "com/example",
+				"com");
+
+			assertThat(jar.getPackages()).contains("com.example.foo.bar")
+				.doesNotContain("com.example.foo", "com.example", "com");
+
+			jar.remove("com/example/foo/bar/x.class");
+			assertThat(jar.getDirectories()).containsKeys("com/example/foo/bar", "com/example/foo", "com/example",
+				"com");
+			assertThat(jar.getPackages()).contains("com.example.foo.bar")
+				.doesNotContain("com.example.foo", "com.example", "com");
+
+			jar.remove("com/example/foo/bar/package-info.class");
+			assertThat(jar.getDirectories()).containsKeys("com/example/foo/bar", "com/example/foo", "com/example",
+				"com");
+			assertThat(jar.getPackages()).doesNotContain("com.example.foo.bar", "com.example.foo", "com.example",
+				"com");
+
+		}
+	}
+
+	@Test
 	public void testWriteFolder() throws Exception {
 		try (Builder b = new Builder()) {
 			b.setIncludeResource("/a/b.txt;literal='ab', /a/c.txt;literal='ac', /a/c/d/e.txt;literal='acde'");
@@ -140,6 +179,7 @@ public class JarTest extends TestCase {
 		}
 	}
 
+	@Test
 	public void testNoManifest() throws Exception {
 		try (Jar jar = new Jar("dot")) {
 			jar.setManifest(new Manifest());
@@ -149,14 +189,16 @@ public class JarTest extends TestCase {
 			ByteArrayOutputStream bout = new ByteArrayOutputStream();
 			jar.write(bout);
 
-			Jar jin = new Jar("dotin", new ByteArrayInputStream(bout.toByteArray()));
-			Resource m = jin.getResource("META-INF/MANIFEST.MF");
-			assertNull(m);
-			Resource r = jin.getResource("a/b");
-			assertNotNull(r);
+			try (Jar jin = new Jar("dotin", new ByteArrayInputStream(bout.toByteArray()))) {
+				Resource m = jin.getResource("META-INF/MANIFEST.MF");
+				assertNull(m);
+				Resource r = jin.getResource("a/b");
+				assertNotNull(r);
+			}
 		}
 	}
 
+	@Test
 	public void testManualManifest() throws Exception {
 		try (Jar jar = new Jar("dot")) {
 			jar.setManifest(new Manifest());
@@ -176,6 +218,7 @@ public class JarTest extends TestCase {
 		}
 	}
 
+	@Test
 	public void testRenameManifest() throws Exception {
 		try (Jar jar = new Jar("dot")) {
 			Manifest manifest = new Manifest();
@@ -202,6 +245,7 @@ public class JarTest extends TestCase {
 		}
 	}
 
+	@Test
 	public void testSimple() throws ZipException, IOException {
 		File file = IO.getFile("jar/asm.jar");
 		try (Jar jar = new Jar("asm.jar", file)) {
@@ -232,6 +276,7 @@ public class JarTest extends TestCase {
 		}
 	}
 
+	@Test
 	public void testNewLine() throws Exception {
 		try (Jar jar = new Jar("dot")) {
 			Manifest manifest = new Manifest();
@@ -257,6 +302,7 @@ public class JarTest extends TestCase {
 		}
 	}
 
+	@Test
 	public void testZipSlip() throws Exception {
 		assertThat(catchThrowable(() -> {
 			try (Jar jar = new Jar(new File("jar/zip-slip.zip"))) {
@@ -266,6 +312,7 @@ public class JarTest extends TestCase {
 			.isInstanceOfAny(IOException.class, UncheckedIOException.class);
 	}
 
+	@Test
 	public void testCreateZipSlip() throws Exception {
 		try (Jar jar = new Jar("zipzlip")) {
 			assertThat(catchThrowable(() -> {

--- a/biz.aQute.bndlib/src/aQute/bnd/osgi/Jar.java
+++ b/biz.aQute.bndlib/src/aQute/bnd/osgi/Jar.java
@@ -326,7 +326,11 @@ public class Jar implements Closeable {
 		} else if (path.equals(Constants.MODULE_INFO_CLASS)) {
 			moduleAttribute = null;
 		}
-		Map<String, Resource> s = directories.computeIfAbsent(getParent(path), dir -> {
+		String dir = getParent(path);
+		Map<String, Resource> s = directories.get(dir);
+		if (s == null) {
+			s = new TreeMap<>();
+			directories.put(dir, s);
 			// make ancestor directories
 			for (int n; (n = dir.lastIndexOf('/')) > 0;) {
 				dir = dir.substring(0, n);
@@ -334,8 +338,7 @@ public class Jar implements Closeable {
 					break;
 				directories.put(dir, null);
 			}
-			return new TreeMap<>();
-		});
+		}
 		boolean duplicate = s.containsKey(path);
 		if (!duplicate || overwrite) {
 			resources.put(path, resource);

--- a/biz.aQute.bndlib/src/aQute/bnd/osgi/Jar.java
+++ b/biz.aQute.bndlib/src/aQute/bnd/osgi/Jar.java
@@ -987,7 +987,8 @@ public class Jar implements Closeable {
 	public List<String> getPackages() {
 		check();
 		return MapStream.of(directories)
-			.filterValue(Objects::nonNull)
+			.filterValue(mdir -> Objects.nonNull(mdir) && !mdir
+				.isEmpty())
 			.keys()
 			.map(k -> k.replace('/', '.'))
 			.collect(toList());

